modules/calib3d/src/solvepnp.cpp

```diff
@@ -110,6 +111,18 @@ void drawFrameAxes(InputOutputArray image, InputArray cameraMatrix, InputArray d
     std::vector<Point2f> imagePoints;
     projectPoints(axesPoints, rvec, tvec, cameraMatrix, distCoeffs, imagePoints);
 
+    cv::Rect imageRect(0, 0, img.cols, img.rows);
+    bool allIn = true;
+    for (size_t i = 0; i < imagePoints.size(); i++)
+    {
+        allIn &= imageRect.contains(imagePoints[i]);
+    }
+
+    if (!allIn)
+    {
+        CV_LOG_WARNING(NULL, "Some of projected axes endpoints are out of frame. The drawn axes may be not relaible.");
```

Comment:
> relaible

Small typo. I am not a native english speaker, my proposition:

--> "Some of the projected axes endpoints are outside of the image bounds. The drawn axes may not be reliable." (accurate / correct instead?)

---

I would add in the message something like this:
- "You may want to pass a zero-distortion coefficient vector to avoid this issue." (or NULL?)

This should be the simplest workaround since most likely `drawFrameAxes` would be used for debugging / display purposes.

---

Maybe another approach would be?
- `projectPoints` without distortion
- clip the coordinates inside the image rectangle
- add distortion
- draw the frame axes

Reply:
> "You may want to pass a zero-distortion coefficient vector to avoid this issue." (or NULL?)

It generates incorrect result for cameras with visible distortion. Axes will be shifted from expected location.